source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -80,13 +80,17 @@ def canStart(cls):
 	def __init__(self):
 		super(VisionEnhancementProvider, self).__init__()
 		Magnification.MagInitialize()
-		Magnification.MagShowSystemCursor(False)
 		Magnification.MagSetFullscreenColorEffect(TRANSFORM_BLACK)
+		Magnification.MagShowSystemCursor(False)
 
 	def terminate(self):
-		super(VisionEnhancementProvider, self).terminate()
-		Magnification.MagShowSystemCursor(True)
-		Magnification.MagUninitialize()
+		try:
+			super(VisionEnhancementProvider, self).terminate()
+			Magnification.MagUninitialize()
+		except:  # noqa: E722
+			raise
+		finally:
+			Magnification.MagShowSystemCursor(True)

```

Comment:
I always try to make the order of __init__ the reverse of terminate. Now, in both __init_- and terminate, we either initialize/unlitialize the engine and then call ShowSystemCursor. IOI would have expected that the latter failed, as you already called MagUNitialize before calling MagShowSYstemCursor.

I intended something like this, if you agree:

```
	def terminate(self):
		try:
			super().terminate()  # Feel free to make this a new style super call
		finally:
			Magnification.MagShowSystemCursor(True)
			Magnification.MagUninitialize()
```

Reply:
Of course, my bad, it's much better that way.